include/swift/ABI/Metadata.h

```diff
@@ -2125,6 +2125,16 @@ struct TargetExtendedExistentialTypeMetadata
     return this->template getTrailingObjects<ConstTargetPointer<Runtime, void>>();
   }
 
+public:
+  /// Project the value pointer from an extended existential container of the
+  /// type described by this metadata.
+  const OpaqueValue *projectValue(const OpaqueValue *container) const;
+
+  OpaqueValue *projectValue(OpaqueValue *container) const {
+    return const_cast<OpaqueValue *>(
+        projectValue((const OpaqueValue *)container));
+  }
```

Comment:
I believe the input type is this way because all the `*ExistentialContainer` types don't share a common supertype that we could use as a more refined upper bound.

As for this abstraction, I'm not sure what the problem with it is. It just abstracts over the different ways to grab the value from the container types, which I need for both the caster and the uwnrapper.